### PR TITLE
Juhlee/bvt2

### PR DESCRIPTION
**Pull request by nicepat**
Added new test cases under BVT. Validated the menu UI. Still need to validate test script execution and calling mechanism.

### CODE REVIEW

File: Testscripts/Windows/RHUI-BVT-VERIFY-RPM-DOWNLOAD.ps1 [hunk 1]
```diff
@@ -0,0 +1,71 @@
+﻿<#-------------Create Deployment Start------------------#>
```
Comment (iamshital) [inline Testscripts/Windows/RHUI-BVT-VERIFY-RPM-DOWNLOAD.ps1]: Remove this file.

File: Testscripts/Windows/RDFELibs.psm1 [hunk 2]
```diff
@@ -0,0 +1,7110 @@
+#region Azure RDFE Methods
```
Comment (iamshital) [inline Testscripts/Windows/RDFELibs.psm1]: Remove this entire file.

File: Testscripts/Linux/DiskHotAddRemove-Serial.ps1 [hunk 3]
```diff
@@ -0,0 +1,104 @@
+Import-Module .\TestLibs\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Linux/DiskHotAddRemove-Serial.ps1]: Libraries are imported by parent scripts now. No need to import here explicitly. (Same goes for all the PS1 test scripts)

File: XML/TestCases/BVT.xml [hunk 4]
```diff
@@ -0,0 +1,494 @@
+<TestCases>
+    <test>
+        <testName>BVT-VERIFY-DEPLOYMENT-PROVISION</testName>
+        <testScript></testScript>
+        <testScriptPs1>BVT-VERIFY-DEPLOYMENT-PROVISION.ps1</testScriptPs1>
```
Comment (iamshital) [inline XML/TestCases/BVT.xml]: <testScriptPs1> is wrong tag. Replace with <PowershellScript>

Comment (iamshital) [reply XML/TestCases/BVT.xml]: testScriptPs1 is wrong tag. Replace with PowershellScript

File: Testscripts/Windows/RDFELibs.psm1 (hunk 2 above)
Comment (nicepat) [reply Testscripts/Windows/RDFELibs.psm1]: Done

File: Testscripts/Linux/DiskHotAddRemove-Serial.ps1 (hunk 3 above)
Comment (nicepat) [reply Testscripts/Linux/DiskHotAddRemove-Serial.ps1]: All 38 PS1 files are updated with this removal

File: Testscripts/Windows/BVT-CAPTURE-VHD.ps1 [hunk 5]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-CAPTURE-VHD.ps1]: Remove this line

File: Testscripts/Windows/BVT-HVMODULES-CHECK.ps1 [hunk 6]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-HVMODULES-CHECK.ps1]: Remove this line

File: Testscripts/Windows/BVT-LAST-CONSOLE-CHECK.ps1 [hunk 7]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-LAST-CONSOLE-CHECK.ps1]: Remove this line

File: Testscripts/Windows/BVT-MTAB-ENTRY-CHECK.ps1 [hunk 8]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-MTAB-ENTRY-CHECK.ps1]: Remove this line

File: Testscripts/Windows/BVT-PASSWD-POLICY-TEST.ps1 [hunk 9]
```diff
@@ -1,4 +1,3 @@
-<#-------------Create Deployment Start------------------#>
 Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-PASSWD-POLICY-TEST.ps1]: Remove this line

File: Testscripts/Windows/BVT-RESOURCE-DISK-FILESYSTEM.ps1 [hunk 10]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-RESOURCE-DISK-FILESYSTEM.ps1]: Remove this line

File: Testscripts/Windows/BVT-ROOT-DEVICE-TIMEOUT-CHECK.ps1 [hunk 11]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-ROOT-DEVICE-TIMEOUT-CHECK.ps1]: Remove this line

File: Testscripts/Windows/BVT-SWAP-CHECK.ps1 [hunk 12]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-SWAP-CHECK.ps1]: Remove this line

File: Testscripts/Windows/BVT-VERIFY-BASH-HISTORY.ps1 [hunk 13]
```diff
@@ -1,4 +1,3 @@
-<#-------------Create Deployment Start------------------#>
 Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-VERIFY-BASH-HISTORY.ps1]: Remove this line

File: Testscripts/Windows/BVT-VERIFY-BOOT-ERROR-WARNINGS.ps1 [hunk 14]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-VERIFY-BOOT-ERROR-WARNINGS.ps1]: Remove this line

File: Testscripts/Windows/BVT-VERIFY-DISKSIZE.ps1 [hunk 15]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-VERIFY-DISKSIZE.ps1]: Remove this line

File: Testscripts/Windows/BVT-VERIFY-FIREWALL-STATUS.ps1 [hunk 16]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-VERIFY-FIREWALL-STATUS.ps1]: Remove this line

File: Testscripts/Windows/BVT-VERIFY-KVP-DAEMON-STATUS.ps1 [hunk 17]
```diff
@@ -1,10 +1,4 @@
-﻿# This script will deploy a VM and verify that the KVP daemon is started at the boot of the VM.
-#
-# Author: Sivakanth Rebba
-# Email	: v-sirebb@microsoft.com
-###################################################################################
-<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-VERIFY-KVP-DAEMON-STATUS.ps1]: Remove this line

File: Testscripts/Windows/BVT-VERIFY-MNT-RESOURCE-README.ps1 [hunk 18]
```diff
@@ -1,4 +1,3 @@
-<#-------------Create Deployment Start------------------#>
 Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-VERIFY-MNT-RESOURCE-README.ps1]: Remove this line

File: Testscripts/Windows/BVT-VERIFY-MNT-RESOURCE-WRITABLE.ps1 [hunk 19]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-VERIFY-MNT-RESOURCE-WRITABLE.ps1]: Remove this line

File: Testscripts/Windows/BVT-VERIFY-NO-ERROR-IN-LOGS.ps1 [hunk 20]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-VERIFY-NO-ERROR-IN-LOGS.ps1]: Remove this line

File: Testscripts/Windows/BVT-VERIFY-PRIVATEKEY-AUTHENTICATION.ps1 [hunk 21]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-VERIFY-PRIVATEKEY-AUTHENTICATION.ps1]: Remove this line

File: Testscripts/Windows/BVT-VERIFY-RESTART.ps1 [hunk 22]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-VERIFY-RESTART.ps1]: Remove this line

File: Testscripts/Windows/BVT-VERIFY-SHUTDOWN.ps1 [hunk 23]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-VERIFY-SHUTDOWN.ps1]: Remove this line

File: Testscripts/Windows/BVT-VERIFY-SSHD-CONFIG.ps1 [hunk 24]
```diff
@@ -1,4 +1,3 @@
-<#-------------Create Deployment Start------------------#>
 Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-VERIFY-SSHD-CONFIG.ps1]: Remove this line

File: Testscripts/Windows/BVT-VERIFY-UUID-FSTAB.ps1 [hunk 25]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-VERIFY-UUID-FSTAB.ps1]: Remove this line

File: Testscripts/Windows/BVT-VERIFY-VERBOSE-ENABLED-LOGS.ps1 [hunk 26]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-VERIFY-VERBOSE-ENABLED-LOGS.ps1]: Remove this line

File: Testscripts/Windows/BVT-VERIFY-VHD-PREREQUISITES.ps1 [hunk 27]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-VERIFY-VHD-PREREQUISITES.ps1]: Remove this line

File: Testscripts/Windows/BVT-WA-VER-CHECK.ps1 [hunk 28]
```diff
@@ -1,5 +1,4 @@
-﻿<#-------------Create Deployment Start------------------#>
-Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
+﻿Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/BVT-WA-VER-CHECK.ps1]: Remove this line

File: Testscripts/Windows/RHUI-BVT-VERIFY-RPM-SOURCE.ps1 [hunk 29]
```diff
@@ -1,4 +1,3 @@
-<#-------------Create Deployment Start------------------#>
 Import-Module .\Testscripts\Windows\RDFELibs.psm1 -Force
```
Comment (iamshital) [inline Testscripts/Windows/RHUI-BVT-VERIFY-RPM-SOURCE.ps1]: Remove this line

File: Testscripts/Linux/DiskHotAddRemove-Serial.ps1 (hunk 3 above)
Comment (nicepat) [reply Testscripts/Linux/DiskHotAddRemove-Serial.ps1]: Done

File: XML/TestCases/BVT.xml (hunk 4 above)
Comment (nicepat) [reply XML/TestCases/BVT.xml]: Can you clarify what you meant?

Comment (nicepat) [reply XML/TestCases/BVT.xml]: Done

**Pull request merged**